cxx/clean_relation.hh

```diff
@@ -132,7 +132,6 @@ class CleanRelation : public Relation<T> {
           // It's safe to unincorporate this element since no other data point
           // refers to it.
           data_r.at(name).erase(items[i]);
-          domains[i]->unincorporate(items[i]);
```

Comment:
I'm pretty sure this is a bug, since domains can be shared across relations (so just because the domain doesn't take a given value in this relation doesn't mean it can't in other relations). (We had resurrected the `unincorporate` code from an obsolete commented-out version.)

Reply:
The garbage collection is now done in irm.incorporate. I added a comment to `Relation::unincorporate` to explain.